plutus-chain-index/README_marconi.md

```diff
@@ -0,0 +1,159 @@
+
+This is an initial draft for the Marconi README file. Going forward, we will need to discuss and refine this document and/or the implementation so that they match.
+
+# Marconi
+
+## The Cardano blockchain indexer for dApp developers
+
+A lightweight customizable solution for indexing and querying the Cardano blockchain.
+
+### Built by IOG in Haskell
+
+* dApp developers can index whatever is important to them. 
+* Marconi is an indexing solution for developers who need to store on-chain data in a local database. 
+
+## Introduction
+
+Marconi is a blockchain query solution that you can configure for the types of information that you need to index and according to how you need to structure your database schema. 
+
+Marconi consists of these three components: 
+
+| Component | Description | 
+| --------- | ----------- | 
+| Streamer  | Provides event stream of configured data types | 
+| Index     | Indexes and stores the data in a database | 
+| Query     | Fetches information from the database |
+
+The streamer attaches to a node, providing an event stream of the configured data types. Your applications can use streams directly. Or streams may be passed to the index component which indexes and stores the data in a database. You can use the query component to fetch information from the database.
+
+## Design Principles
+
+### High Performance through Customized Solutions for Each Use Case
+
+The philosophy behind Marconi is to maintain high performance by customizing the solution for each use case. You can filter the streams of information to transmit only the data you need for your dApp, minimizing network traffic and database storage. You can also customize database schemas and queries so that the dApps have the most efficient and scalable application-specific API to interact with the data.
+
+### Handling Blockchain Rollback Events Properly
+
+Marconi is designed to enable dApps to handle blockchain rollback events properly. Since the head of the chain is constantly advancing and rolling back as the consensus algorithm advances the chain, this causes challenging synchronization issues for dApps that need rapid updates of the chain state. You can obtain basic notifications of chain advances and rollbacks from the streaming component which wraps the application data types with the event type that indicates the current slot index for the block. Rollback events will have lower slot numbers that can be checked by the application which can then invalidate any obsolete state it has cached.
```

Comment:
> Rollback events will have lower slot numbers that can be checked by the application which can then invalidate any obsolete state it has cached.

In the current design events provided by plutus-streaming can be of two types: `RollForward` or `RollBack`. Consumers of the stream do not need to look at the slot numbers to differenciate.

Reply:
I have removed the statement about lower slot numbers being checked.